src/pevent_lib.cc

```diff
@@ -55,11 +58,34 @@ static void display_system_config(void) {
   }
 }
 
+int pevent_compute_min_mmap_order(int min_buffer_size_order,
+                                  uint32_t sample_stack_user) {
+  int ret_order = min_buffer_size_order;
+  // perf events and allocation events should be roughtly the same size
+  size_t single_event_size =
+      sample_stack_user - 1 + sizeof(ddprof::AllocationEvent);
+  // Ensure we can at least fit 8 samples within one buffer
+  while ((perf_mmap_size(ret_order) / single_event_size) <
```

Comment:
`perf_mmap_size` returns `(2^order+1) * PAGE_SIZE`, but one page is used for metadata, hence the usable size is `perf_mmap_size(ret_order)-PAGE_SIZE`.
Note that before this PR, ring buffer could not hold 8 samples: ring buffer usable size was 262144 and 8 samples would be `(4096*8+perf_sample_data)*8`

Reply:
Indeed. 8 sounds like such a low number...
I would be in favour of increasing min size to order 7. This could come with more mmap failures though.

Reply:
Why not slightly decrease the default sample stack size to make sure that 8 samples fit into the buffer ?

Reply:
I'll adjust the log message to help users with this.

Reply:
Good point, adjusted.